string object
---
string[] grasp_poses
